packages/dev/core/src/Materials/Node/nodeMaterial.ts

```diff
@@ -2347,10 +2349,37 @@ export class NodeMaterial extends PushMaterial {
                             if (!skipBuild) {
                                 nodeMaterial.build();
                             }
-                            resolve(nodeMaterial);
                         } catch (err) {
                             reject(err);
                         }
+
+                        // Ensures all textures are ready to render.
+                        const textureReadyPromises: Promise<void>[] = [];
+                        if (waitForTextureReadyness) {
+                            nodeMaterial.getActiveTextures().forEach((texture) => {
+                                const internalTexture = texture.getInternalTexture();
+                                if (internalTexture && !internalTexture.isReady) {

```

Comment:
nope but you are totally right and it could have been a great catch !!! 

https://github.com/BabylonJS/Babylon.js/blob/master/packages/dev/core/src/Materials/Textures/internalTexture.ts#L85

Reply:
Ahah, I've been had!

It's a bit like the observer being called `onLoadedObservable` in `InternalTexture` and `onLoadObservable` in `Texture`!